src/main/java/fi/aalto/cs/apluscourses/intellij/actions/AboutAction.java

```diff
@@ -2,10 +2,11 @@
 
 import com.intellij.openapi.actionSystem.AnAction;
 import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.project.DumbAware;
 import fi.aalto.cs.apluscourses.ui.AboutDialog;
 import org.jetbrains.annotations.NotNull;
 
-public class AboutAction extends AnAction {
+public class AboutAction extends AnAction implements DumbAware {
```

Comment:
This `DumbAware` thing is something that I have been thinking I should study more about because I guess there are some other parts of our code that should be `DumbAware`. If you have robust understanding how does it work and how it should be used, I'd be keen to get some insights. :)

Reply:
Implementing `DumbAware` makes the action available during indexing (that long process that happens when you start the IDE). As long as the action doesn't access the indices, it can/should implement `DumbAware`. The problem is that I'm not 100% sure which methods from the API need the indices and which don't, but the good news is that an exception is thrown if the indices are accessed before they are ready :smile:

Reply:
I created a new ticket of this #129